enhancements/update/cvo-log-level-api.md

```diff
@@ -0,0 +1,325 @@
+
+This enhancement proposes to add a new field to configure the log level of the CVO
+by modifying the `ClusterVersion` resource.
+Because more additional CVO specific options may be exposed by the
+`ClusterVersion` in the future, this enhancement proposes to add a new
+`OperatorOptions` field to the existing [`ClusterVersionSpec`](https://github.com/openshift/api/blob/3e5de946111c67cb1f2b09ff7da569012d15931d/config/v1/types_cluster_version.go#L50) struct:
+
+```go
+type ClusterVersionSpec struct {
+...
+	// operatorOptions contains options to configure the CVO.
+	//
+	// +optional 
+	OperatorOptions *ClusterVersionOperatorOptions `json:"operatorOptions,omitempty"`
+}
+```
+
+The new `OperatorOptions` field will group all the CVO specific fields
+to make the `ClusterVersionSpec` more organized, future-proof, and clear.
+The `ClusterVersionOperatorOptions` type is defined as:
+
+```go
+// ClusterVersionOperatorOptions contains options to configure the CVO.
+//
+// +k8s:deepcopy-gen=true
+type ClusterVersionOperatorOptions struct {
+	// operatorLogLevel controls the verbosity level of logs produced by the CVO.
+	//
+	// Valid values are: "Normal", "Debug", "Trace", "TraceAll".
+	// Defaults to "Normal".
+	// +optional
+	// +kubebuilder:default=Normal
+	LogLevel CVOLogLevel `json:"logLevel,omitempty"`
+}
+```
+
+The `CVOLogLevel` type is defined as:
+
+```go
+// CVOLogLevel represents the verbosity level of logs produced by the CVO.
+//
+// +kubebuilder:validation:Enum="";Normal;Debug;Trace;TraceAll
+type CVOLogLevel string
+
+var (
+	// CVOLogLevelNormal is the default.  Normal, working log information, everything is fine, but helpful notices for auditing or common operations.  In kube, this is probably glog=2.
+	CVOLogLevelNormal CVOLogLevel = "Normal"
+
+	// CVOLogLevelDebug is used when something went wrong.  Even common operations may be logged, and less helpful but more quantity of notices.  In kube, this is probably glog=4.
+	CVOLogLevelDebug CVOLogLevel = "Debug"
+
+	// CVOLogLevelTrace is used when something went really badly and even more verbose logs are needed.  Logging every function call as part of a common operation, to tracing execution of a query.  In kube, this is probably glog=6.
+	CVOLogLevelTrace CVOLogLevel = "Trace"
+
+	// CVOLogLevelTraceAll is used when something is broken at the level of API content/decoding.  It will dump complete body content.  If you turn this on in a production cluster
+	// prepare from serious performance issues and massive amounts of logs.  In kube, this is probably glog=8.
+	CVOLogLevelTraceAll CVOLogLevel = "TraceAll"
+)
+```
+
+Ideally, this enhancement would propose to use the existing [`LogLevel`](https://github.com/openshift/api/blob/36ce464529eb357673342c06be5886c5463cfc50/operator/v1/types.go#L91-L107)
+type that is used by `ClusterOperators` defined in the [`github.com/openshift/api/operator/v1`](https://github.com/openshift/api/tree/master/operator/v1) package.
```

Comment:
I don't think this is `ClusterOperators` but more various `Operator` type CRDs defined each for specific use cases.

The CVO should probably evolve an operator group CRD for its configuration at this level to be consistent

Reply:
I greatly appreciate the fast review, Joel. Let me go over the operator group, and I'll get back to you.

Reply:
A new operator group CRD is proposed in the new changes. Thank you for the suggestions.